library/ecdh.c

```diff
@@ -91,64 +102,161 @@ int mbedtls_ecdh_compute_shared( mbedtls_ecp_group *grp, mbedtls_mpi *z,
  */
 void mbedtls_ecdh_init( mbedtls_ecdh_context *ctx )
 {
+#if defined(MBEDTLS_ECDH_LEGACY_CONTEXT)
     memset( ctx, 0, sizeof( mbedtls_ecdh_context ) );
+
+    mbedtls_ecp_group_init( &ctx->grp );
+    mbedtls_ecp_point_init( &ctx->Q   );
+    mbedtls_ecp_point_init( &ctx->Qp  );
```

Comment:
We were previously just memsetting the context to 0 for the legacy context. Now, we memset the context to 0 and also call these init functions, while with the non-legacy context we simply memset to 0. Is this the right way around? Shouldn't the legacy context configuration behave identically to before, and the new context configuration call the individual init functions?

Reply:
It doesn't make any difference, because currently `mbedtls_ecp_group_init()` just calls memset, and `mbedtls_ecp_point_init()` has an equivalent behaviour to memset (initialising the coordinates to `+0` versus `0`).

But you are right, it is better to be more robust and call the `_init()` functions.

Reply:
After your latest changes, we appear to be calling the init functions for both the legacy configuration and the non-legacy configuration.

I was asking if we actually needed to call the `mbedtls_ecp_group_init()` and `mbedtls_ecp_point_init()` functions for the legacy configuration, as we didn't before this PR. This PR added them for the legacy configuration.

Reply:
I don't remember why I added it, I think it is better to not change behaviour, I will restore both cases to the old behaviour.

Reply:
I don't think its necessary to the legacy configuration to call the init functions (they weren't called before), but should we call the init functions for the new non-legacy configuration?